src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts

```diff
@@ -3031,10 +3031,13 @@ export async function readTextureToTexelViews(
         }
       `,
     });
+    const colorType = kTextureFormatInfo[texture.format].color?.type;
     const sampleType = isDepthTextureFormat(texture.format)
       ? 'unfilterable-float'
       : isStencilTextureFormat(texture.format)
       ? 'uint'
+      : colorType === 'float'
+      ? 'unfilterable-float'
       : kTextureFormatInfo[texture.format].color?.type ?? 'unfilterable-float';
```

Comment:
nit: there's a lot of logic happening in this ternary chain, I was going to suggest pulling it out into a function but I realized it can be simplified by taking advantage of type narrowing:
```suggestion
    const info = kTextureFormatInfo[texture.format];
    const sampleType = info.depth
      ? 'unfilterable-float' // depth or depth+stencil (TODO: should comment something about why we pick this sample type in this case)
      : info.stencil
      ? 'uint'
      : info.color.type === 'float'
      ? 'unfilterable-float'
      : info.color.type;
```